File: lib/src/main/java/wang/mycroft/lib/net/OkHttpClientMaker.java
package wang.mycroft.lib.net;

import okhttp3.OkHttpClient;

/**
 * {@link OkHttpClient}构造接口
 *
 * @author Mycroft Wong
 * @date 2019年11月15日
 */
public interface OkHttpClientMaker {

    /**
     * 构造OkHttpClient
     *
     * @return {@link OkHttpClient}
     */
    OkHttpClient makeOkHttpClient();
}

File: lib/src/main/java/wang/mycroft/lib/net/OkHttpClientMaker.kt
package wang.mycroft.lib.net

import okhttp3.OkHttpClient

/**
 * [OkHttpClient]构造器
 *
 * @author Mycroft Wong
 * @date 2019年11月15日
 */
interface OkHttpClientMaker {
    /**
     * 构造[OkHttpClient]
     *
     * @return [OkHttpClient]
     */
    fun makeOkHttpClient(): OkHttpClient
}